### PR TITLE
Vault-env can write files

### LINKED ISSUE #185: TemporaryFilesystem strategy from SystemD howto doesn't work
Hello, in your systemd tutorial, you suggest the usage of `TemporaryFilesystem` to create a temporary tmpfs directory and store the secrets in a file inside it so the executable in `ExecStart=` can read it.

Example:
``` shell
[Service]
TemporaryFileSystem=/private
ExecStartPre=vault-cli get mysecret --output=/private/path/to/secret/file
```

In my case, I want to use this with Postgres, so I can store the TLS certificate and private key inside Vault and save it inside that directory so Postgres can read it from there.

The issue with that strategy is that the private directory created by `TemporaryFilesystem` will be removed after the `ExecStartPre=` call and will not be accessible from `ExecStart=`.

 Do you know if there is some way around it? I know this is more a systemd specific issue, but since it is referenced in your howto, I'm asking here so we can improve the howto by specifying what more is needed to be done for this strategy to work or simply remove it from the howto since it will not work in the end.

More about the systemd issue [here](https://github.com/systemd/systemd/issues/8695).

### PR DESCRIPTION
**Pull request by ewjoachim**
Closes #185 

### Successful PR Checklist:

- [x] Tests (units, integration, 100% coverage)
  - [ ] (not applicable?)
- [x] Documentation
  - [ ] (not applicable?)
- [x] Had a good time contributing?

### CODE REVIEW

File: vault_cli/cli.py [hunk 1]
```diff
@@ -402,16 +403,45 @@ def delete(client_obj: client.VaultClientBase, name: str, key: Optional[str]) ->
...
+    name, --envvar was added later for clarity, both work identically, and neither is
+    formerly deprecated. See command help for details.
+    """,
+)
+@click.option(
+    "--file",
+    multiple=True,
+    help="""
+    Write a secret from this path into a file on the filesystem. Expected format is
+    path/in/vault:key=/path/in/filesystem . This option is meant to be used when you are
```
Comment (ewjoachim) [inline vault_cli/cli.py:423]:
```suggestion
    path/in/vault:key=/path/in/filesystem . This option is meant to be used when
```

**Comment by ewjoachim** [comment]
Thanks @sheb 

Only thing left seems:
```
./vault_cli/cli.py:2:1: F401 'json' imported but unused
./tests/unit/test_cli.py:463:36: F541 f-string is missing placeholders
```

**Comment by ewjoachim** [comment]
Meanwhile: a mypy update has broken the build :D

Your choice: we can solve this in this PR or in a new PR, it's independant from the modification this PR did bring. I have limited time to take care of this now, but I'll review anything you ping me on :)

File: vault_cli/cli.py [hunk 2]
```diff
@@ -465,11 +465,27 @@ def delete(client_obj: client.VaultClientBase, name: str, key: Optional[str]) ->
     formerly deprecated. See command help for details.
     """,
 )
+@click.option(
+    "--file",
+    multiple=True,
+    help="""
+    Write a secret from this path into a file on the filesystem. Expected format is
+    path/in/vault[:key]=/path/in/filesystem . This option is meant to be used when you are
+    your command can only read its inputs from a file and not from the environment (e.g.
```
Comment (ewjoachim) [inline vault_cli/cli.py:474]:
```suggestion
    path/in/vault[:key]=/path/in/filesystem . This option is meant to be used when
    your command can only read its inputs from a file and not from the environment (e.g.
```

**Review by sheb — APPROVED** [review]

**Pull request merged**